omnichain/staking/contracts/Staking.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "@zetachain/protocol-contracts/contracts/zevm/SystemContract.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/zContract.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@zetachain/toolkit/contracts/BytesHelperLib.sol";
+
+contract Staking is ERC20, zContract {
+    error SenderNotSystemContract();
+    error WrongChain();
+    error NotAuthorizedToClaim();
+
+    SystemContract public immutable systemContract;
+    uint256 public immutable chain;
+
+    mapping(address => uint256) public stakes;
+    mapping(address => address) public beneficiaries;
+    mapping(address => uint256) public lastStakeTime;
+    uint256 public rewardRate = 1;
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint256 chainID,
```

Comment:
for consistency use chainID in param name and state variable or chain in both. I prefer chainID the state and chainID_ the param to avoid collision

Reply:
995b953